File: Tugas-PostDay4/src/demo_package/src/subscriberNode.cpp
#include "ros/ros.h"
#include "demo_package/Topic.h"

void callback(const demo_package::Topic::ConstPtr &psn){
    ROS_INFO("\nIngfo Diterima Capt\n");
    ROS_INFO("Umur : %ld\n", psn->usia);
    ROS_INFO("Tinggi : %lf\n", psn->tinggi);
    ROS_INFO("Nama : %s\n", psn->nama.c_str());
    ROS_INFO("Hadir : %s\n", (psn->kehadiran ? "Hadir" : "Tidak Hadir"));
}

int main(int argc, char ** argv){
    ros::init(argc,argv,"subscriberNode");

    ros::NodeHandle nh;

    ros::Subscriber data_subs = nh.subscribe("topikUtama",1000,callback);

    ros::spin();

    return 0;
}

File: Tugas-PostDay4/src/demo_package/script/SubscriberNode.py
#! /usr/bin/env python3
import rospy
from demo_package.msg import Topic

def callback(data):
    # Mencetak pesan yang diterima
    rospy.loginfo("ABSENSI INSTITUT RUSDI BARBERSHOP:")
    rospy.loginfo("Usia: %ld" % data.usia)
    rospy.loginfo("Tinggi: %lf" % data.tinggi)
    rospy.loginfo("Nama: %s" % data.nama)
    rospy.loginfo("Kehadiran: %s" % ("Hadir" if data.kehadiran else "Tidak Hadir"))

def pendengar():
    # Inisialisasi Node
    rospy.init_node("subscriber",anonymous=True)

    # Inisialisasi Subscriber
    rospy.Subscriber("mainTopic",Topic,callback)

    rospy.spin()

if __name__ == "__main__":
    pendengar()